sklearn/utils/tests/test_extmath.py

```diff
@@ -687,18 +691,56 @@ def rng():
 
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
-def test_incremental_weighted_mean_and_variance_simple(rng, dtype):
+@pytest.mark.parametrize("as_list", (True, False))
+def test_incremental_weighted_mean_and_variance_simple(rng, dtype, as_list):
     mult = 10
     X = rng.rand(1000, 20).astype(dtype) * mult
     sample_weight = rng.rand(X.shape[0]) * mult
-    mean, var, _ = _incremental_mean_and_var(X, 0, 0, 0, sample_weight=sample_weight)
+    mean, var, _ = _incremental_mean_and_var(
+        X.tolist() if as_list else X, 0, 0, 0, sample_weight=sample_weight
```

Comment:
If we keep this test case I'd suggest rewriting this to move the `if` outside of the function call. I had to stare at it quite a bit to parse all the commas. Now that I have seen it, it is "clear" - but I think whoever comes to this test in the future trying to understand why it is broken will appreciate it if we make it easy to parse the code. Something like
```python
X = X.tolist() if as_list else X
mean, var, _ = _incremental_mean_and_var(X, 0, 0, 0, sample_weight=sample_weight)
```
What do you think?

Reply:
Sure!